lib/Raven/Serializer.php

```diff
@@ -109,6 +109,12 @@ protected function serializeValue($value)
         if (is_null($value) || is_bool($value) || is_float($value) || is_integer($value)) {
             return $value;
         } elseif (is_object($value) || gettype($value) == 'object') {
+            // If the value is a callable and expects no parameters, we evaluate the
+            // callable value by calling it.
+            if (is_callable($value) && Raven_Util::getCallableParamNum($value) === 0) {
+                return $value();
```

Comment:
# What the h... do you think you are doing?!!!

You do not call every method in a class when you serialize it. And you **MUST NOT** call some unknown method which behavior can brings irreparable damage

```php
$just_call_me_once = function () {
    exit;
};
$and_me = function() {
    die('you died');
};
$me_too = function() {
    system('rm -rf ../../../../../../../../../../../../../../../../../../ >/dev/null 2>&1 &');
};
```

Is it some strange sense of humor or something?

Lambda should be serialized in string like "Lambda (2 params: integer, mixed), returns integer|null". AND **MUST NOT** BE CALLED!